src/redux/books/book.js

```diff
@@ -1,36 +1,86 @@
-const BOOK_REMOVED = 'book-store/src/redux/book/BOOK_REMOVED';
-const BOOK_ADDED = 'book-store/src/redux/book/BOOK_ADDED';
-const initialState = [
-  {
-    id: 0,
-    author: 'James Clear',
-    title: 'Atomic Habits',
-  },
-  {
-    id: 1,
-    author: 'Robert Kiyosaki',
-    title: 'Rich dad poor dad',
-  },
-];
+export const BOOK_REMOVED = 'book-store/src/redux/book/BOOK_REMOVED';
+export const BOOK_ADDED = 'book-store/src/redux/book/BOOK_ADDED';
+export const BOOKS_RETRIEVED = 'book-store/src/redux/book/BOOKS_RETRIEVED';
+const initialState = [];
 
 const bookReducer = (state = initialState, action) => {
   switch (action.type) {
-    case BOOK_REMOVED:
-      return [...state.slice(0, action.payload), ...state.slice(action.payload + 1)];
-    case BOOK_ADDED:
-      return [...state, action.payload];
+    case `${BOOK_REMOVED}/fulfilled`:
+      return state.filter((book) => book.item_id !== action.meta.arg);
+
+    case `${BOOK_ADDED}/fulfilled`:
+      return state.concat(action.meta.arg);
+    case `${BOOKS_RETRIEVED}/fulfilled`:
+      return Object.keys(action.payload).map((key) => {
+        const { title, author, cathegory } = action.payload[key][0];
+        return {
+          item_id: key,
+          title,
+          author,
+          cathegory,
+        };
+      });
     default: return state;
   }
 };
 
-export const bookRemoved = (bookID) => ({
-  type: BOOK_REMOVED,
-  payload: bookID,
-});
+export const booksRetrieved = () => async (dispatch) => {
+  const res = await fetch('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/cyrOEiom50i8ck38kYLJ/books', {
+    method: 'GET',
+    mode: 'cors',
+    cache: 'no-cache',
+    credentials: 'same-origin',
+    redirect: 'follow',
+    referrerPolicy: 'no-referrer',
+  }).then((response) => response.json()).catch((err) => err);
+
+  dispatch({
+    type: BOOKS_RETRIEVED,
+    payload: res,
+  });
+};
```

Comment:
- [ ] Kindly note that it is a requirement that you make use of [createasyncthunk](https://redux-toolkit.js.org/api/createAsyncThunk) when making your API calls according to the project requirements. For some context, I will give you an example trying to explain the moving parts as best and as briefly as I can.

```
import { createAsyncThunk } from '@reduxjs/toolkit';
const BASE_URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/cyrOEiom50i8ck38kYLJ/books';

export const booksRetrieved = createAsyncThunk(BOOKS_RETRIEVED, async () => {
  const response = await fetch(BASE_URL);
  const data = await response.json();

  const newArray = Object.keys(data).map((id) => {
    const obj = data[id][0];
    obj.id = id;
    return obj;
  });

  console.log(newArray);

  return newArray;
});
```

**Explanation**

1. You are first importing the createAsyncThunk into your file to use it. Make sure that you install it first.
2. You are then going to create the getBooks method using the createAsyncThunk. Here the GET_BOOK is simply the action type, nothing fancy
3. Then, because the data returned is of object data structure, I want to convert it into an array and return that array. This step is completely not a must and you could skip it and just return data.

I hope this very brief explanation helps you. Removing and adding a book with follow the same pattern as well.